### DISCUSSION TITLE
Github vs Azure Devops for Azure based system?

### DISCUSSION THREAD

**Discussion by pranavq212** [github/india#64] [Satellite India 2021 Q&A] [answered]
I want to choose a new Devops solution for an Azure PAAS based solution, source control will be git based & mainly using c# for development with some scripting language. 
I want to check what are the advantages of Github over Azure Devops or Vice versa for comparison based on performance, ease of development, compatibility with Azure and integration with new features.
I am ready to go for paid subscription for both services. I am not quite sure how to do this comparison. Can some one help with any resources for this or link to any similar comparison done earlier by someone else?

Any insight however small in correct direction will be helpful.

**Comment by martinwoodward** [comment, marked as answer]
If you already have both and you are starting today, then personally I would go with GitHub.  And while I'm biased, I also wrote the book on Azure DevOps - well, 6 of them actually, but don't like to boast ;)

GitHub has more integrations (with Azure but also with other cloud hosted providers), great VS integration for the C# development and then you also have access to the security features as well as the bigger GitHub Actions community.

**Reply by pranavq212** [reply]
Thanks @martinwoodward for the prompt reply. That's what I was thinking while thinking of Cloud provider agnostic solution to avoid vendor lock-in. But was not sure about support of GitHub for Azure since Azure DevOps for sure has good support for Azure anyway.
Is there good GitHub reference to start for implementing CI/CD for Azure PAAS solutions for learning & POC. I'm trying to do same things as POC on both platforms to analyse for ease of learning, support & performance. I want to have something concrete to make use case for choosing one over other.
Cost although is not primarily factor but certainly comes into perspective if both matches quite close.

**Reply by msupernaut** [reply]
hey @pranavq212 - try these out :- 
https://lab.github.com/githubtraining/github-actions:-continuous-delivery-with-azure
https://lab.github.com/MSUSDEV/build-end-to-end-cicd-capabilities-directly-in-github

**Reply by lyraddigital** [reply]
Best part about using Github. It's highly supported and is always growing at a fast rate. It's also got great people to reach out to if you need help with anything.

**Reply by pranavq212** [reply]
@msupernaut thanks a lot. I will definitely try on that
